### DISCUSSION TITLE
collapse by default?

### DISCUSSION THREAD

**Discussion by bdelamatre** [laravel/nova-issues#3887] [Ideas & Feature Requests] [answered]
Is there a setting to collapse menu items and/or relations by default? I know that my browser will remember these settings, but there are a number of menu groups and resources that I would prefer collapsed by default.

**Comment by davidhemphill** [comment]
There's currently no setting for this, since it was designed to be a user-specific preference.

**Reply by ziming** [reply]
@davidhemphill 

How about a setting where if it's the user 1st time viewing that Resource detail page, allow us to set which relations expand or collapse by default?

But on subsequent visits, it follows the user preferences saved in the browser for that Resource.

**Reply by jeanfrancoisgb** [reply]
Would absolutely love that feature. A small quality of life improvement that goes a long way.

**Reply by marijoo** [reply]
Especially for heavy or rarely relevant relations this would be a really helpful feature. Something like this:

```php
MorphMany::make('Meta')->collapsedByDefault();
```

**Reply by szabizs** [reply]
I back this feature also.

**Reply by saschaende** [reply]
Yes please add collapsedByDefault()

**Reply by kusab85** [reply]
Yes, I would definitely love this feature too. Both for menu groups and resource relations.

**Comment by pedro-santiago** [comment]
@davidhemphill just a single liner on Collapsable.js. May substitute with `this.item.collapsedByDefault` as suggested. 
```
  created() {
    this.collapsed = JSON.parse(localStorage.getItem(this.localStorageKey)) ?? true;
  },
```

**Reply by pedro-santiago** [reply]
This way it works as intended (user preference) and gives us some flexibility to set the defaults. In my scenario I have a large sidebar, so if everything is not collapsed, it scrolls and gets ugly.

**Comment by davidhemphill** [comment, marked as answer]
🚀 In the next release, you will be able to set `MenuSection`, `MenuGroup`, `HasMany`, `HasManyThrough`, `MorphToMany`, `BelongsToMany`, and `Panel` as `collapsedByDefault`. This will allow you to set the initial collapsed state if not already set by the user's preference.

**Comment by didix16** [comment]
Hi @davidhemphill , I'm using nova 4.22.2 version and it seems collapsedByDefault() is not collpasing by default the panel. It only shows an arrow to collapse but it is not collapsed. Is that something that could be fixed? Thanks in advance

**Reply by szabizs** [reply]
It is working without issues, did you set collapsible() ?

**Reply by didix16** [reply]
Well, I'm using collapsedByDefault() method that seems is calling collapsable at the same time.
![imagen](https://user-images.githubusercontent.com/10291092/223978732-3a42f782-9a8c-4ac1-ab59-06179fd5c27d.png)

**Reply by crynobone** [reply]
Submit a bug report with full reproducing code and description if you have any issues: https://github.com/laravel/nova-issues/issues/new?assignees=&labels=&template=1_Bug_report.md
